packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -0,0 +1,51 @@
+import { Event, getEventJson } from './event';
+import { CfnDetectorModel } from './iotevents.generated';
+
+/**
+ * Properties for defining a state of a detector
+ */
+export interface StateProps {
+  /**
+   * The name of the state
+   */
+  readonly stateName: string;
+
+  /**
+   * Specifies the actions that are performed when the state is entered and the `condition` is `TRUE`
+   *
+   * @default None
+   */
+  readonly onEnterEvents?: Event[];
+}
+
+/**
+ * Defines a state of a detector
+ */
+export class State {
+  /**
+   * The name of the state
+   */
+  public readonly stateName: string;
+
+  constructor(private readonly props: StateProps) {
+    this.stateName = props.stateName;
+  }
+
+  /**
+   * Return the state property JSON
+   */
+  public toStateJson(): CfnDetectorModel.StateProperty {
```

Comment:
Can we make this method `@internal`? I don't think there's a need to expose it in the public API of this module.

It will require renaming it to `_toStateJson()`.